LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/TaskExtensions.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.NetworkServer
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Threading.Tasks;
+
+    internal static class TaskExtensions
+    {
+        public static IList<Exception> GetExceptions(this IList<Task> tasks)
+        {
+            var result = new List<Exception>();
+
+            foreach (var task in tasks)
+            {
+                if (!task.IsCompleted)
+                {
+                    throw new InvalidOperationException("All tasks must have completed.");
+                }
+
+                if (task.IsCompletedSuccessfully) continue;
+
+                if (task.IsCanceled && task.TryGetCanceledException(out var ex))
+                    result.Add(ex);
+
+                if (task.IsFaulted && task.Exception is { } someException)
+                {
+                    result.Add(someException.InnerExceptions switch
+                    {
+                        { Count: 1 } => someException.InnerExceptions[0],
+                        _ => someException
+                    });
+                }
```

Comment:
Some patterns can be merged and the switch expression simplified to a conditional:

```suggestion
                if (task is { IsFaulted: true, Exception: { } aggregateException })
                    result.Add(aggregateException.InnerExceptions is { Count: 1 } exceptions ? exceptions[0] : aggregateException);
```